### PR TITLE
Add WebSubHub Spec to the repository

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#2162](https://github.com/ballerina-platform/ballerina-standard-library/issues/2162)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
+# Specification: Ballerina WebSubHub Library
+
+_Owners_: @shafreenAnfar @chamil321 @ayeshLK    
+_Reviewers_: @shafreenAnfar    
+_Created_: 2021/11/23  
+_Updated_: 2021/11/23  
+_Issue_: [#2162](https://github.com/ballerina-platform/ballerina-standard-library/issues/2162)
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:7]: I think this issue is more relevant than the above. Maybe we can update the issue with a comment saying final design has evolved from the original to the current one.

https://github.com/ballerina-platform/ballerina-standard-library/issues/786

File: docs/spec/spec.md [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+# Introduction
+
+This is the specification for WebSubHub standard library, an API framework to implement WebSub compliant `hub` services 
+and `publisher` clients using [Ballerina programming language](https://ballerina.io/).
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:12]: For the sake of consistency, lets stick to this `Introduction`

https://github.com/ballerina-platform/module-ballerina-jwt/blob/master/docs/spec/spec.md

File: docs/spec/spec.md (hunk 1 above)
Comment (ayeshLK) [reply docs/spec/spec.md]: Fixed with [eb6cf1df06216511ce28c833508ef1c300649bcf](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/378/commits/eb6cf1df06216511ce28c833508ef1c300649bcf)

File: docs/spec/spec.md [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+   * 2.1. [Hub Listener](#21-hub-listener)
+   * 2.2. [Hub Service](#22-hub-service)
+     * 2.2.1. [Service Annotation](#221-service-annotation)
+3. [Hub Client](#3-hub-client)
+4. [Publisher Client](#4-publisher-client)
+
+## 1. Overview
+
+[WebSub](https://www.w3.org/TR/websub/) is a real-time content delivery protocol over HTTP(s) and it is a simple 
+specification which evolved from [PubSubHubbub](https://github.com/pubsubhubbub/PubSubHubbub).
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:27]: It should HTTP(S) not HTTP(s).

File: docs/spec/spec.md [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+
+WebSub specification describes three main roles: 
+- Publisher: An implementation that advertises a topic and hub URL on one or more resource URLs.
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:30]: I think we can remove `An implementation that` and just start with what follows. i.e. `Publisher: Advertises a topic and hub URL on one or more resource URLs.`

File: docs/spec/spec.md [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+- Subscriber: An implementation that discovers the `hub` and topic URL given a resource URL, subscribes to updates at 
+the `hub`, and accepts content distribution requests from the `hub`.
+- Hub: An implementation that handles subscription requests and distributes the content to subscribers when the 
+corresponding topic URL has been updated.
+
+`WebSubHub` is an API framework which is derived from the WebSub specification which could be used by developers to
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:36]: framework should be library.

File: docs/spec/spec.md [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+            certFile: "./resources/server.crt",
+            keyFile: "./resources/server.key"
+        }
+    }
+);
+```
+
+### 2.2 Hub Service
+
+`websubhub:Service` is responsible to handle the received events. Underlying `http:Service` will receive the original
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:81]: isn't `responsible for handling` ?

File: docs/spec/spec.md [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+In addition to that, following remote methods are optional,
+- onSubscription
+- onSubscriptionValidation
+- onUnsubscritpion
+- onUnsubscriptionValidation
+
+In the event of a bad request from the `publisher` or the `subscriber`, the WebSubHub dispatcher will automatically send 
+back the appropriate response to the client.
+
+Following is a sample implementation of the `hub`:
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:152]: I don't think we need this part.

File: docs/spec/spec.md [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+};
+```
+
+While the below remote methods are strictly WebSub compliant,
+- onSubscription 
+- onSubscriptionValidation
+- onSubscriptionIntentVerified 
+- onUnsubscritpion 
+- onUnsubscriptionValidation
+- onUnsubscriptionIntenVerified
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:133]: I think we should briefly explain the functionally of each remote method.

File: docs/spec/spec.md (hunk 3 above)
Comment (ayeshLK) [reply docs/spec/spec.md]: Fixed with [e9ec5bccc217575bffcb58e22d9f2bb4edfcef94](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/378/commits/e9ec5bccc217575bffcb58e22d9f2bb4edfcef94)

File: docs/spec/spec.md (hunk 4 above)
Comment (ayeshLK) [reply docs/spec/spec.md]: Fixed with [e9ec5bccc217575bffcb58e22d9f2bb4edfcef94](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/378/commits/e9ec5bccc217575bffcb58e22d9f2bb4edfcef94)

File: docs/spec/spec.md (hunk 5 above)
Comment (ayeshLK) [reply docs/spec/spec.md]: Fixed with [e9ec5bccc217575bffcb58e22d9f2bb4edfcef94](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/378/commits/e9ec5bccc217575bffcb58e22d9f2bb4edfcef94)

File: docs/spec/spec.md (hunk 8 above)
Comment (ayeshLK) [reply docs/spec/spec.md:322]: Fixed with [19f7114601a4bb48d08efdf5331d55779437e8bb](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/378/commits/19f7114601a4bb48d08efdf5331d55779437e8bb)

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**